framework/docs/source/how-to-use-json-output.rst

```diff
@@ -0,0 +1,166 @@
+Use JSON outputs for Flower CLIs
+================================
+
+The `Flower CLIs <ref-api-cli.html>`_ come with a built-in JSON output mode. This mode
+is useful when you want to consume the output of a Flower CLI programmatically. For
+example, you might want to use the output of the ``flwr`` CLI in a script or a
+continuous integration pipeline.
+
+To enable JSON output, simply pass the ``--format json`` flag to the CLI. In this guide,
+we'll show you how to use JSON output with the ``flwr run``, ``ls``, and ``stop``
+commands.
+
+.. |flwr_run| replace:: ``flwr run``
+
+.. |flwr_ls| replace:: ``flwr ls``
+
+.. |flwr_stop| replace:: ``flwr stop``
+
+.. _flwr_ls: ref-api-cli.html#flwr-ls
+
+.. _flwr_run: ref-api-cli.html#flwr-run
+
+.. _flwr_stop: ref-api-cli.html#flwr-stop
+
+``flwr run`` JSON output
+------------------------
+
+The |flwr_run|_ command runs a Flower app. By default, the command prints the status of
+the app build and run process as follows:
+
+.. code-block:: bash
+
+    $ flwr run
+    Loading project configuration...
+    Success
+    🎊 Successfully built flwrlabs.myawesomeapp.1-0-0.014c8eb3.fab
+    🎊 Successfully started run 1859953118041441032
+
+To get the output in JSON format, simply pass the ``--format json`` flag:
+
+.. code-block:: bash
+
+    $ flwr run --format json
+    {
+      "success": true,
+      "run-id": 1859953118041441032,
+      "fab-id": "flwrlabs/myawesomeapp",
+      "fab-name": "myawesomeapp",
+      "fab-version": "1.0.0",
+      "fab-hash": "014c8eb3",
+      "fab-filename": "flwrlabs.myawesomeapp.1-0-0.014c8eb3.fab"
+    }
+
+The JSON output contains the following fields:
+
+- ``success``: A boolean indicating whether the command was successful.
+- ``run-id``: The ID of the run.
+- ``fab-id``: The ID of the Flower app.
+- ``fab-name``: The name of the Flower app.
+- ``fab-version``: The version of the Flower app.
+- ``fab-hash``: The hash of the Flower app.
+- ``fab-filename``: The filename of the Flower app.
+
+If the command fails, the JSON output will contain two fields, ``success`` with the
+value of ``false`` and ``error-message``. For example, if the command fails to find the
+name of the federation on the SuperLink, the output will look like this:
+
+.. _json_error_output:
+
+.. code-block:: bash
+
+    $ flwr run --format json
+    {
+      "success": false,
+      "error-message": "Loading project configuration... \nSuccess\n There is no `[missing]` federation declared in the `pyproject.toml`.\n The following federations were found:\n\nfed-existing-1\nfed-existing-2\n\n"
+    }
+
+``flwr ls`` JSON output
+-----------------------
+
+The |flwr_ls|_ command lists all the runs in the current project. By default, the
+command list the details of one provided run ID or all runs in a Flower federation in a
+tabular format:
```

Comment:
```suggestion
The |flwr_ls|_ command lists all the runs in the current project. By default, the
command lists the details all runs in a Flower federation in a tabular format:
```

Reply:
I'm not sure if mentioning the ID topic is needed (given the examples shown below)

Reply:
I've removed this mention to the last paragraph of this section. I think it's still worth mentioning what the JSON output will be for one entry.